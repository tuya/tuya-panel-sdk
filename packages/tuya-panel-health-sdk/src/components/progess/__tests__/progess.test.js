import React from 'react';
import { View } from 'react-native';
import { shallow } from 'enzyme';
import Progess from '../index';

describe('Progess components', () => {
  it('basic render', () => {
    const wrapper = shallow(<Progess percent={12} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('componentWillReceiveProps render', () => {
    let size = 40;
    const wrapper = shallow(<Progess percent={12} />);
    size = 60;
    wrapper.setProps({ size });
    expect(wrapper).toMatchSnapshot();
    let color = '#FFC204';
    wrapper.setProps({ color });
    wrapper.unmount();
  });
});
