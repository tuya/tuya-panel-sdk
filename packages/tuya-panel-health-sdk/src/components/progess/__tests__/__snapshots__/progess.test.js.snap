// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progess components basic render 1`] = `
<View
  style={
    Object {
      "height": 100,
      "width": 100,
    }
  }
>
  <Svg
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "height": "100%",
        "position": "relative",
        "width": "100%",
      }
    }
    viewBox="0 0 100 100"
  >
    <Path
      d="M 50,50 m 0,-46 a 46,46 0 1 1 0,92 a 46,46 0 1 1 0,-92"
      fillOpacity="0"
      stroke="#EEF8FF"
      strokeLinecap="round"
      strokeWidth={8}
    />
    <Path
      d="M 50,50 m 0,-46 a 46,46 0 1 1 0,92 a 46,46 0 1 1 0,-92"
      fillOpacity="0"
      stroke="#56C0FF"
      strokeDasharray={
        Array [
          289.02652413026095,
          289.02652413026095,
        ]
      }
      strokeDashoffset={254.34334123462963}
      strokeLinecap="round"
      strokeWidth={8}
    />
  </Svg>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "position": "absolute",
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
      ]
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      style={
        Object {
          "fontSize": 28,
          "fontWeight": "bold",
        }
      }
      type={null}
      weight={null}
    >
      12
      %
    </TYText>
  </View>
</View>
`;

exports[`Progess components componentWillReceiveProps render 1`] = `
<View
  style={
    Object {
      "height": 60,
      "width": 60,
    }
  }
>
  <Svg
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "height": "100%",
        "position": "relative",
        "width": "100%",
      }
    }
    viewBox="0 0 100 100"
  >
    <Path
      d="M 50,50 m 0,-46 a 46,46 0 1 1 0,92 a 46,46 0 1 1 0,-92"
      fillOpacity="0"
      stroke="#EEF8FF"
      strokeLinecap="round"
      strokeWidth={8}
    />
    <Path
      d="M 50,50 m 0,-46 a 46,46 0 1 1 0,92 a 46,46 0 1 1 0,-92"
      fillOpacity="0"
      stroke="#56C0FF"
      strokeDasharray={
        Array [
          289.02652413026095,
          289.02652413026095,
        ]
      }
      strokeDashoffset={254.34334123462963}
      strokeLinecap="round"
      strokeWidth={8}
    />
  </Svg>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "position": "absolute",
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
      ]
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      style={
        Object {
          "fontSize": 28,
          "fontWeight": "bold",
        }
      }
      type={null}
      weight={null}
    >
      12
      %
    </TYText>
  </View>
</View>
`;
